inc/Engine/Optimization/CSSTrait.php

```diff
@@ -191,6 +195,128 @@ protected function move( ConverterInterface $converter, $content, $source ) {
 		return str_replace( $search, $replace, $content );
 	}
 
+	/**
+	 * Replace local imports with their contents recursively.
+	 *
+	 * @since 3.8.6
+	 *
+	 * @param string $content CSS Content.
+	 * @param string $target Target CSS file path.
+	 *
+	 * @return string
+	 */
+	protected function combine_imports( $content, $target ) {
+		$import_regexes = [
+			// @import url(xxx)
+			'/
+		# import statement
+		@import
+
+		# whitespace
+		\s+
+
+			# open url()
+			url\(
+
+				# (optional) open path enclosure
+				(?P<quotes>["\']?)
+
+					# fetch path
+					(?P<path>.+?)
+
+				# (optional) close path enclosure
+				(?P=quotes)
+
+			# close url()
+			\)
+
+			# (optional) trailing whitespace
+			\s*
+
+			# (optional) media statement(s)
+			(?P<media>[^;]*)
+
+			# (optional) trailing whitespace
+			\s*
+
+		# (optional) closing semi-colon
+		;?
+
+		/ix',
+
+			// @import 'xxx'
+			'/
+
+		# import statement
+		@import
+
+		# whitespace
+		\s+
+
+			# open path enclosure
+			(?P<quotes>["\'])
+
+				# fetch path
+				(?P<path>.+?)
+
+			# close path enclosure
+			(?P=quotes)
+
+			# (optional) trailing whitespace
+			\s*
+
+			# (optional) media statement(s)
+			(?P<media>[^;]*)
+
+			# (optional) trailing whitespace
+			\s*
+
+		# (optional) closing semi-colon
+		;?
+
+		/ix',
+		];
+
+		// find all relative imports in css.
+		$matches = [];
+		foreach ( $import_regexes as $import_regexe ) {
+			if ( preg_match_all( $import_regexe, $content, $regex_matches, PREG_SET_ORDER ) ) {
+				$matches = array_merge( $matches, $regex_matches );
+			}
+		}
```

Comment:
```suggestion
			if ( ! preg_match_all( $import_regexe, $content, $matches, PREG_SET_ORDER ) ) {
				return $content;
			}
```

Wouldn't this be enough to achieve the same result? We already get an array of matches from `preg_match_all()`, I'm not sure why it would be needed to merge with another empty array.

Reply:
We have two regex patterns here to match the following schema:-
1. `@import url(xxx)`
2. `@import 'xxx'`

So I'm looping on the array of regexes `$import_regexes` and getting matches then add them to this new array.
What do you think?